### PR TITLE
docs: (mostly) add motivation docs

### PR DESCRIPTION
**Pull request by kentcdodds**
Also update some styling stuff and other random things to make things
nice. Please merge :)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -14,5 +14,8 @@
     "import/no-unresolved": "off",
     "import/no-unassigned-import": "off",
     "eol-last": "off", // the crowdin stuff doesn't include a line ending 🙄
+  },
+  "globals": {
+    "Prism": false,
```
Comment (kentcdodds) [inline .eslintrc]: Weird things happened with highlighting if I tried to import Prism, so I'll just use the global :(

File: components/docs-page.js [hunk 2]
```diff
@@ -122,7 +122,14 @@ function DocSection(props) {
           dangerouslySetInnerHTML={{__html: mdToHTMLUnwrapped(title)}}
         />
       </Anchor>
-      <h4 dangerouslySetInnerHTML={{__html: mdToHTMLUnwrapped(subtitle)}} />
+      <glamorous.Div
```
Comment (kentcdodds) [inline components/docs-page.js]: It's not really semantically accurate to use an `h4` and it looks not awesome. So I formatted this a little differently in a way I think is nicer

Comment (paulmolluzzo) [inline components/docs-page.js]: A `div` doesn't seem semantically accurate either. 😕 Maybe don't use `mdToHTMLUnwrapped` so we at least get a `p` tag? I could be missing something though...

Comment (kentcdodds) [inline components/docs-page.js]: Oh, good idea. I'll do that.

**Pull request merged**